config/e2e/batch_job.yaml

```diff
@@ -33,7 +33,7 @@ spec:
       containers:
         - name: e2e
           image: {{ .E2EImage }}
-          imagePullPolicy: IfNotPresent
+          imagePullPolicy: Always
```

Comment:
I'm not too familiar with how E2EImage was populated. Were we reusing image names across test runs and so they were getting cached even if they changed?

Reply:
Image caching is a problem if you manually run the tests with uncommitted changes. 

I kept the setting because it won't affect the normal CI runs (E2E image name is unique and will require a pull from the registry anyway).

Reply:
I think we may have introduced `IfNotPresent` a while ago in order to run tests on Minikube. We didn't have a registry there, so we used to build the image in Minikube context. Meaning it was possible to use it, but not to pull it from Minikube.
The situation is different now, we do run a registry in Minikube if I remember correctly (`hack/registry.sh`).